### PR TITLE
[fix](struct-type) fix be core when load array orc file

### LINKED ISSUE #16925: [Bug] [daily] be core when load array orc file
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master: doris-0.0.0-trunk-2074b83

### What's Wrong?

jenkins failed case:
https://ci-builds.apache.org/job/Doris/job/doris_daily_enable_vectorized/213/testReport/(root)/test_array_load/test_array_orc/

core file:
image

be.out
https://ci-builds.apache.org/job/Doris/job/doris_daily_enable_vectorized/213/testReport/(root)/test_sys_check_core/test_check_be_core/

### What You Expected?

be ok

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zxealous**
# Proposed changes

Issue Number: close #16925

## Problem summary
in be/src/vec/data_types/data_type_factory.cpp:168, col_desc.contains_nulls[0] memory access out of bounds

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/format/orc/vorc_reader.cpp [hunk 1]
```diff
@@ -607,21 +607,28 @@ TypeDescriptor OrcReader::_convert_to_doris_type(const orc::Type* orc_type) {
     case orc::TypeKind::TIMESTAMP_INSTANT:
         return TypeDescriptor(PrimitiveType::TYPE_DATETIMEV2);
     case orc::TypeKind::LIST: {
+        // Use Object PrimitiveType to construct TypeDescriptor also need init containers_nulls with the default value of bool
         TypeDescriptor list_type(PrimitiveType::TYPE_ARRAY);
         list_type.children.emplace_back(_convert_to_doris_type(orc_type->getSubtype(0)));
```
Comment (cambyzju) [inline be/src/vec/exec/format/orc/vorc_reader.cpp:612]: better add a function like:
`TypeDescriptor::add_sub_type(type, bool is_nullable = true);`

contains_nulls should be true as default.

File: be/src/vec/exec/format/orc/vorc_reader.cpp [hunk 2]
```diff
@@ -607,21 +607,28 @@ TypeDescriptor OrcReader::_convert_to_doris_type(const orc::Type* orc_type) {
...
+        list_type.contains_nulls.push_back(false);
```
Comment (xy720) [inline be/src/vec/exec/format/orc/vorc_reader.cpp:613]: should be true as default

Comment (zxealous) [reply be/src/vec/exec/format/orc/vorc_reader.cpp:613]: ok

File: be/src/vec/exec/format/orc/vorc_reader.cpp (hunk 1 above)
Comment (zxealous) [reply be/src/vec/exec/format/orc/vorc_reader.cpp:612]: ok

File: be/src/runtime/types.cpp [hunk 3]
```diff
@@ -270,6 +270,15 @@ TypeDescriptor::TypeDescriptor(const google::protobuf::RepeatedPtrField<PTypeNod
     }
 }
 
+void TypeDescriptor::add_sub_type(TypeDescriptor&& sub_type, std::string&& field_name,
+                                  bool&& is_nullable) {
+    children.emplace_back(sub_type);
+    if (!field_name.empty()) {
```
Comment (xy720) [inline be/src/runtime/types.cpp:276]: Better use a new method add_field_name for field_name.
Because field_name is only used for struct type.

**Review by xy720 — APPROVED** [review]
LGTM

**Review by zxealous — COMMENTED** [review]
run buildall

**Comment by zxealous** [comment]
run buildall

**Pull request merged**